lib/srv/desktop/rdp/rdpclient/src/client.rs

```diff
@@ -304,6 +327,93 @@ impl Client {
         })
     }
 
+    async fn handle_clipboard(
+        cgo_handle: CgoHandle,
+        write_stream: &mut RdpWriteStream,
+        x224_processor: Arc<Mutex<X224Processor>>,
+        clipboard: &mut Option<String>,
+        clipboard_function: ClipboardFunction,
+    ) -> ClientResult<()> {
+        match clipboard_function {
+            ClipboardFunction::RequestFormatList => {
+                Client::process_cliprdr(x224_processor, write_stream, |c| c.initiate_copy(&[]))
+                    .await?
+            }
+            ClipboardFunction::Update(data) => {
+                clipboard.replace(data);
+                Client::process_cliprdr(x224_processor, write_stream, |c| {
+                    c.initiate_copy(&[ClipboardFormat::new(ClipboardFormatId::CF_UNICODETEXT)])
+                })
+                .await?;
+            }
+            ClipboardFunction::RemoteCopy(data) => {
+                let format = data
+                    .iter()
+                    .find(|cf| {
+                        cf.id() == ClipboardFormatId::CF_UNICODETEXT
+                            || cf.id() == ClipboardFormatId::CF_TEXT
+                    })
+                    .map(ClipboardFormat::id);
+                if let Some(cf) = format {
+                    Client::process_cliprdr(x224_processor, write_stream, move |c| {
+                        c.initiate_paste(cf)
+                    })
+                    .await?;
+                }
+            }
+            ClipboardFunction::FormatDataRequest(req) => {
+                let data = clipboard
+                    .as_ref()
+                    .filter(|_| req == ClipboardFormatId::CF_UNICODETEXT)
+                    .map(|s| util::to_unicode(s.as_str(), true))
+                    .map_or_else(FormatDataResponse::new_error, FormatDataResponse::new_data);
+                Client::process_cliprdr(x224_processor, write_stream, |c| {
+                    c.submit_format_data(data)
+                })
+                .await?;
+            }
+            ClipboardFunction::FormatDataResponse(data) => unsafe {
+                let mut data = if data.ends_with(&[0u8, 0u8]) {
```

Comment:
It looks like we've lost format support here compared to what we have

https://github.com/gravitational/teleport/blob/e19bfee84116c17390847a92f65dae92c6afbfb1/lib/srv/desktop/rdp/rdpclient/src/cliprdr.rs#L276-L287

Reply:
Not really, we support (and ask for) text only, and you can distinguish between CF_UNICODE and CF_TEXT without checking the format you asked for - unicode always ends up with `0u16` = `0u8, 0u8`